datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -86,6 +86,9 @@ use datafusion_physical_expr_common::physical_expr::fmt_sql;
 use futures::{ready, Stream, StreamExt, TryStreamExt};
 use parking_lot::Mutex;
 
+const HASH_JOIN_SEED: RandomState =
```

Comment:
maybe left some comments here to explain this change
```suggestion
/// Hard-coded seed to ensure hash values from the hash join differ from `RepartitionExec`, avoiding collisions.
const HASH_JOIN_SEED: RandomState =
```